packages/firebase_admob/lib/firebase_admob.dart

```diff
@@ -184,6 +184,84 @@ class InterstitialAd extends MobileAd {
   Future<bool> load() => _doLoad("loadInterstitialAd");
 }
 
+/// [RewardedVideoAd] status changes reported to [RewardedVideoAdListener]s.
+///
+/// The [rewarded] event is particularly important, since it indicates that the
+/// user has watched a video to completion and should be given an in-app reward.
+enum RewardedVideoAdEvent {
+  loaded,
+  failedToLoad,
+  opened,
+  leftApplication,
+  closed,
+  rewarded,
+  started,
+}
+
+/// Signature for a [RewardedVideoAd] status change callback. The optional
+/// parameters are only used when the [RewardedVideoAdEvent.rewarded] event
+/// is sent, and will be null for all others.
+typedef void RewardedVideoAdListener(RewardedVideoAdEvent event,
+    [String rewardType, int rewardAmount]);
+
+/// The AdMob rewarded video ad. The AdMob API uses a singleton for its rewarded
+/// video ads, and this class is designed to match.
+///
+/// Apps should assign a callback function to [RewardedVideoAd]'s listener
+/// property in order to receive reward notifications from the AdMob SDK:
+/// ```
+/// RewardedVideoAd.instance.listener = (RewardedVideoAdEvent event,
+///     [String rewardType, int rewardAmount]) {
+///     print("You were rewarded with $rewardAmount $rewardType!");
+///   }
+/// };
+/// ```
+///
+/// The function will be invoked when any of the events in
+/// [RewardedVideoAdEvent] occur.
+///
+/// To load and show ads, call the load method:
+/// ```
+/// RewardedVideoAd.instance.load(myAdUnitString, myTargetingInfoObj);
+/// ```
+///
+/// Later (any point after your listener callback receives the
+/// RewardedVideoAdEvent.loaded event), call the show method:
+/// ```
+/// RewardedVideoAd.instance.show();
+/// ```
+///
+/// Only one rewarded video ad can be loaded at a time. Because the creatives
+/// are so large, it's a good idea to start loading an ad well in advance of
+/// when it's likely to be needed.
+class RewardedVideoAd {
+  static final RewardedVideoAd _instance = new RewardedVideoAd.private();
+
+  RewardedVideoAd.private();
```

Comment:
I tried using a constant constructor, but received an error (either when building or in `flutter analyze`) because the listener property is not also constant. I've revised based on everything else about your suggestion, but let me know if other changes are needed.